code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -757,33 +775,21 @@ butcher
 	var/mob/living/critter/flock/drone/F = holder.owner
 	if(!F?.flock)
 		return
-	for(var/atom/T in view(target_range, holder.owner))
-		//handle vehicles first
-		if (isvehicle(T))
-			var/vehicle_is_enemy = F.flock.isEnemy(T) //if someone gets in an enemy pod, attack
-			var/enemy_found = FALSE
-			for (var/mob/occupant in T)
-				if (!F.flock.isEnemy(occupant) && !vehicle_is_enemy)
+	for(var/atom/T in F.flock.enemies)
+		if(IN_RANGE(T,holder.owner,target_range))
```

Comment:
Won't this result in drones trying to shoot people through walls? You could switch to `if (T in holder.owner.view)` which would probably still be faster than iterating over view.

Reply:
yeah, that would probably be better.